pkg/proxy/iptables/proxier.go

```diff
@@ -798,11 +818,13 @@ func getLocalIPs(endpointsMap proxyEndpointsMap) map[types.NamespacedName]sets.S
 	for svcPortName := range endpointsMap {
 		for _, ep := range endpointsMap[svcPortName] {
 			if ep.isLocal {
-				nsn := svcPortName.NamespacedName
-				if localIPs[nsn] == nil {
-					localIPs[nsn] = sets.NewString()
+				if ip := ep.IPPart(); ip != "" {
```

Comment:
Why would this case happen?  If legit, leave a comment?

Reply:
I suppose this depends on how defensive we want to be with our coding. Here ep.IPPart() could return a null string if net.SplitHostPort() detects bad syntax and returns a null string (along with an error). This can happen (theoretically) if the string in ep.endpoint is not quite right. For example, if the ep.endpoint has too many colons, e.g. "1.2.3.4::9999", or brackets that are out of place, e.g. "1.2.3.4:[0]", then net.SplitHostPort() returns a null string and an error. It's probably very unlikely that we'd end up with a bad format in ep.endpoint at this point, with sanity checks that are done elsewhere. I'll leave this null-string check in for now and add a comment, let me know if you feel this is being too defensive.